packages/devtools/lib/src/memory/memory_service.dart

```diff
@@ -60,7 +60,8 @@ class InboundReferences extends Response {
 
 class InboundReference extends Response {
   InboundReference._fromJson(Map<String, dynamic> json) {
-    //source = json['source'],
+    //source = json['source']
+    // For vm_service_lib 3.21.1, use ['ObjRef']?
```

Comment:
@terrylucas do you happen to be using any other private RPCs directly like this? As of 3.23 `InboundReference` will be part of the public protocol and this will break. Any remaining private RPCs that are being used should be roughly documented in  `service_undocumented.md` in `vm_service_lib` so we can use the generated code rather than creating a custom implementation that will need to be fixed later.

Reply:
No that's the only one left.